packages/cli/src/modulars/moduleLoader.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {existsSync} from 'fs';
+import path from 'path';
+import {NETWORK_FAMILY} from '@subql/common';
+import {networkPackages} from './config';
+import {ModuleCache} from './types';
+
+const moduleCache: Partial<ModuleCache> = {};
+
+export function loadDependency<N extends NETWORK_FAMILY>(network: N): ModuleCache[N] {
```

Comment:
I think we should add some runtime checks here to make sure that the package has the properties defined by `INetworkCommonModule`. Then we can have an error like `${packageName} is not compatible, please make sure you're on the latest version`